appbuilder/services/npm-service.ts

```diff
@@ -75,6 +79,54 @@ export class NpmService implements INpmService {
 		}).future<void>()();
 	}
 
+	public search(projectDir: string, keywords: string[], args?: string[]): IFuture<IBasicPluginInformation[]> {
+		return ((): IBasicPluginInformation[] => {
+			let result: IBasicPluginInformation[] = [];
+			let spawnResult = this.executeNpmCommand(projectDir, args || [], keywords.join(" ")).wait();
+			if (spawnResult.stderr) {
+				// npm will write "npm WARN Building the local index for the first time, please be patient" to the stderr and if it is the only message on the stderr we should ignore it.
+				let splitError = spawnResult.stderr.split("\n");
+				if (splitError.length > 2 || splitError[0].indexOf("Building the local index for the first time") === -1) {
```

Comment:
The comment above states that if `Building the local index...` is the **only**error message we should ignore it, but then why is the condition `splitError.length > 2` instead of `splitError.length > 1`? If I get two errors and the first is `Building the local index...` then I won't actually get an error.